tensorflow/c/c_api.cc

```diff
@@ -262,7 +262,7 @@ static Status TF_StringDecode_Impl(const char* src, size_t src_len,
   if (p == nullptr) {
     return InvalidArgument("invalid string encoding or truncated src buffer");
   }
-  if (len64 > std::numeric_limits<size_t>::max()) {
+  else if (len64 > std::numeric_limits<size_t>::max()) {
```

Comment:
Why this change? The previous `if` block contains an unconditional `return`, so why does the `else` here help?